components/button/button.js

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { Pressable } from "react-native";
+import buttonStyles from '@/styles/buttons';
+
+type PropsType = { style: Object };
+
+export const PrimaryButton = (props: PropsType): React$Element<Pressable> => {
+    const { style, children, ...passThroughProps } = props;
+    const buttonStyle = {
+        ...buttonStyles.primaryButton,
+        ...style
+    }
+    return (<Pressable { ...passThroughProps } style={ [buttonStyle] }>{children}</Pressable>);
```

Comment:
Pressable is the newer future proof replacement for TouchableOpacity.
https://reactnative.dev/docs/pressable